hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -2508,7 +2495,18 @@ class Emit[C](
             baos.invoke[Array[Byte]]("toByteArray"),
             stageName,
             mb.getObject(tsd)))
-          decodeResult(cb)
+
+          cb.assign(len, encRes.length())
+          val pt = PCanonicalArray(bodySpec.encodedType.decodedSType(bodySpec.encodedVirtualType).asInstanceOf[SBaseStruct].fieldEmitTypes(0).storageType)
+          val resultArray = pt.constructFromElements(cb, region, len, deepCopy = false) { (cb, i) =>
+            cb.assign(ib, bodySpec.buildCodeInputBuffer(Code.newInstance[ByteArrayInputStream, Array[Byte]](encRes(i))))
+            val eltTupled = bodySpec.encodedType.buildDecoder(bodySpec.encodedVirtualType, parentCB)
+              .apply(cb, region, ib)
+              .asBaseStruct
+            eltTupled.loadField(cb, 0)
+          }
+          cb.assign(encRes, Code._null)
```

Comment:
Is it not possible to also free each block after decoding it? As written we'd need twice the RAM to read a given result array, I think.

Reply:
nice